lib/testing_utils.py

```
@@ -0,0 +1,89 @@
+                return function(*args, **kwargs)
+            finally:
+                elapsed: float = time.perf_counter() - start
+                if should_print:
+                    print(f"[{function.__name__}] ({elapsed:.4f}s)\n") 
+
+        return wrapper
+
+    return decorator
+
+@track_runtime(should_print=True)
+def init_model(model_class: type[T]) -> T:
```

Ruff (0.15.7)

[warning] Generic function `init_model` should use type parameters

Use type parameters

(UP047)